### DISCUSSION TITLE
Can I use jotai-devtools in react native apps?

### DISCUSSION THREAD

**Discussion by zerosrat** [jotaijs/jotai-devtools#56] [Q&A] [answered]
link: https://jotai.org/docs/tools/devtools
I explore the source code of the project. I find out the project constructs the UI using `@mantine/core` lib which is only available in web apps. It seems we can't use devtools in rn apps.

**Comment by arjunvegda** [comment, marked as answer]
`<DevTools/>` is web-only. For React Native apps, we currently recommend using Redux DevTools as a debugging solution via the [`useatomsdevtools`](https://jotai.org/docs/tools/devtools#useatomsdevtools).

We'll need to import the hook using a second entry point (without the `@mantine/core`) to do this.

Eg.

```ts
import { useAtomsDevtools } from 'jotai-devtools/utils';

export const DevTools = () => {
  useAtomsDevtools('store-name-here');
  return null;
};
```

If you're using Expo or Metro to bundle your RN app, you may run into issues with this entry point above due to Expo/Metro's experimental support for `export`. [This comment](https://github.com/jotaijs/jotai-devtools/issues/50#issuecomment-1480486726) might help in that case

**Reply by zerosrat** [reply]
Thanks, `useAtomsDevtools` + React Native Debugger works fine with me.

**Comment by imvikysharma** [comment]
Hello there,

jotai-devtools is a development tool for the Jotai state management library that provides a visual representation of the state tree and the ability to modify state values during development.

While Jotai itself can be used in React Native apps, jotai-devtools is designed to work with the React DevTools extension, which is not available in React Native. Therefore, jotai-devtools is not compatible with React Native and cannot be used in a React Native app.

I would like to recommend a book on React Native/JS

https://www.amazon.com/dp/B0BVRKSY6G?maas=maas_adg_A78FB1C22F0F576564CF29D2524311F0_afap_abs&ref_=aa_maas&tag=maas
